src/braket/circuits/circuit.py

```diff
@@ -148,7 +156,7 @@ def __init__(self, addable: AddableTypes | None = None, *args, **kwargs):
         self._parameters = set()
         self._observables_simultaneously_measurable = True
         self._has_compiler_directives = False
-
+        self._qubits = QubitSet()
```

Comment:
It's not clear that this change is an improvement.

You're trading off (some) performance vs. memory (cache the qubits) and code complexity (we now have to make sure that whenever qubits are changed we update this set). I wouldn't do it unless the perf improvement is significant in some use case.

Reply:
I agree this is creating a little bit more complexity for how the set is updated but this will benefit larger circuits. I shall gather some benchmarking data with this branch.

Reply:
Repeated runs for unit-tests has gained around a 5% speed up on my instance.